### PR TITLE
Add test for BTF_KIND_FLOAT

### PR DESCRIPTION
**Pull request by chenhengqi**
Add a test case for BTF_KIND_FLOAT.

### CODE REVIEW

File: libbpf-cargo/src/test.rs [hunk 1]
```diff
@@ -2145,3 +2145,30 @@ pub struct __anon_3 {
 
     assert_definition(&btf, struct_bpf_sock_tuple, expected_output);
 }
+
+#[test]
+fn test_btf_dump_float() {
+    let prog_text = r#"
+#include "vmlinux.h"
```
Comment (anakryiko) [inline libbpf-cargo/src/test.rs:2152]: do you need vmlinux.h at all?

Comment (chenhengqi) [reply libbpf-cargo/src/test.rs]: No, just remove it.

File: libbpf-cargo/src/btf/btf.rs [hunk 2]
```diff
@@ -731,7 +731,7 @@ impl<'a> Btf<'a> {
     fn load_type(&mut self, data: &'a [u8]) -> Result<BtfType<'a>> {
         let t = data.pread::<btf_type>(0)?;
         let extra = &data[size_of::<btf_type>()..];
-        let kind = (t.info >> 24) & 0xf;
```
Comment (anakryiko) [inline libbpf-cargo/src/btf/btf.rs:734]: please check if it's the only place where we extract kind from btf_type. It's probably best to move this into a separate function, just like libbpf does.

Comment (chenhengqi) [reply libbpf-cargo/src/btf/btf.rs:734]: OK, will follow up on a new PR.

Currently, the code already has a `get_kind()` method, but it is used to extract kind_flag actually.

**Comment by insearchoflosttime** [comment]
I'm seeing this test failure in CI:

```
error: test failed, to rerun pass '-p libbpf-cargo --lib'
---- test::test_btf_dump_float stdout ----
Metadata for package=proj
	null
Found bpf progs to compile:
	UnprocessedObj { package: "proj", path: "/tmp/.tmpyV83Wg/proj/src/bpf/prog.bpf.c", out: "/tmp/.tmpyV83Wg/proj/target/bpf", name: "prog" }
Building /tmp/.tmpyV83Wg/proj/src/bpf/prog.bpf.c
thread 'test::test_btf_dump_float' panicked at 'assertion failed: `(left == right)`
  left: `"f32"`,
 right: `"std::ffi::c_void"`', libbpf-cargo/src/test.rs:2161:5
```

@chenhengqi  would you mind taking a look?

**Comment by chenhengqi** [comment]
> I'm seeing this test failure in CI:
> 
> ```
> error: test failed, to rerun pass '-p libbpf-cargo --lib'
> ---- test::test_btf_dump_float stdout ----
> Metadata for package=proj
> 	null
> Found bpf progs to compile:
> 	UnprocessedObj { package: "proj", path: "/tmp/.tmpyV83Wg/proj/src/bpf/prog.bpf.c", out: "/tmp/.tmpyV83Wg/proj/target/bpf", name: "prog" }
> Building /tmp/.tmpyV83Wg/proj/src/bpf/prog.bpf.c
> thread 'test::test_btf_dump_float' panicked at 'assertion failed: `(left == right)`
>   left: `"f32"`,
>  right: `"std::ffi::c_void"`', libbpf-cargo/src/test.rs:2161:5
> ```
> 
> @chenhengqi would you mind taking a look?

The test passed in my dev machine.
And the CI logs shows that it does NOT triggered on the latest commit. Just force push to trigger a rerun.

**Comment by insearchoflosttime** [comment]
I'll update the clang version CI uses to clang-13 (right now it's on clang-10). Some older clang versions emit void for float types.

**Comment by insearchoflosttime** [comment]
The git workflow uses ubuntu-latest (ubuntu 20.04) which does not support clang-13 yet. We will have to put the test on hold until then. We can merge in the first commit (fixing the btf kind extraction) but I think it'll have to be resubmitted as a standalone PR without the test in order to be merged in

**Comment by chenhengqi** [comment]
How about just ignore the test for now ?

**Comment by insearchoflosttime** [comment]
Great, I just merged this in. Thanks for contributing to libbpf-rs @chenhengqi!

**Pull request merged**